packages/gatsby/cache-dir/app.js

```diff
@@ -164,6 +165,20 @@ apiRunnerAsync(`onClientEntry`).then(() => {
 
       renderer(<Root />, rootElement, () => {
         apiRunner(`onInitialClientRender`)
+
+        // Render query on demand overlay
+        if (process.env.GATSBY_QUERY_ON_DEMAND_LOADING_INDICATOR) {
```

Comment:
this is not the same check we have in https://github.com/gatsbyjs/gatsby/commit/d3dd3e15e27ea040d764a81c6f77f15c281f2499

Reply:
best to just `git cherry-pick -x d3dd3e15e27ea040d764a81c6f77f15c281f2499` (commit being the commit that is one from `master` branch - not single commit from PR)

Reply:
👍 done